src/systems/physics/Physics.cc

```diff
@@ -3808,12 +3821,33 @@ void PhysicsPrivate::UpdateCollisions(EntityComponentManager &_ecm)
             contactMsg->mutable_collision2()->set_name(
               removeParentScope(scopedName(collEntity2, _ecm, "::", 0), "::"));
           }
-          for (const auto &contact : contactData)
+          for (const auto &contact_data : contactData)
           {
             auto *position = contactMsg->add_position();
-            position->set_x(contact->point.x());
-            position->set_y(contact->point.y());
-            position->set_z(contact->point.z());
+            position->set_x(contact_data.first->point.x());
+            position->set_y(contact_data.first->point.y());
+            position->set_z(contact_data.first->point.z());
+
```

Comment:
We need to check that the extra contact data, which would be `contact_data.second`, is not `nullptr` before accessing it because it's optional. There might be some physics engines that don't expose that data.